Source/Peripheral.swift

```diff
@@ -543,3 +547,25 @@ extension Peripheral: Equatable { }
 public func == (lhs: Peripheral, rhs: Peripheral) -> Bool {
     return lhs.peripheral == rhs.peripheral
 }
+
+fileprivate extension Peripheral {
+    fileprivate typealias `Self` = Peripheral
+
+    /**
+     Filters an item list based on the provided UUID list. Only items returned whose UUID matches an item in the provided UUID list.
+     Each UUID should have at least one item matching in the items list. Otherwise the result is nil.
+
+     - uuids: a UUID list or nil
+     - items: items to be filtered
+     - itemUUID: closure to extract the UUID from the items
+     - Returns: the filtered item list
+    */
+    static func filterCachedItems<T>(uuids: [CBUUID]?, items: [T], itemUUID: (T) -> CBUUID) -> [T]? {
```

Comment:
Maybe instead of T which is unconstrained we could constrain in to some protocol(names are not thought through)
```swift
protocol UUIDIdentifiable {
  var uuid: CBUUID { get }
}
extension CBCharacteristic: UUIDIdentifiable { }
extension CBService: UUIDIdentifiable { }
```
What about making it global? It's not associated with the `Peripheral` in any way.
Thanks to this we could have one less parameter in function call and we avoid closures.
Then interface of `filterUUIDItems` method would look like:
```swift
func filterUUIDItems<T: UUIDIdentifiable>(uuids: [CBUUID]?, items: [T]) -> [T]? {
```